tiled/server/core.py

```diff
@@ -22,36 +22,32 @@
 from .. import queries
 from ..adapters.mapping import MapAdapter
 from ..queries import KeyLookup, QueryValueError
-from ..structures import node  # noqa: F401
+from ..serialization import register_builtin_serializers
 from ..utils import (
     APACHE_ARROW_FILE_MIME_TYPE,
     SerializationError,
     UnsupportedShape,
-    modules_available,
     safe_json_dump,
 )
 from . import schemas
 from .etag import tokenize
 from .utils import record_timing
 
 del queries
-if modules_available("numpy", "dask.array"):
-    from ..structures import array as _array  # noqa: F401
-
-    del _array
-if modules_available("pandas", "pyarrow", "dask.dataframe"):
-    from ..structures import dataframe as _dataframe  # noqa: F401
-
-    del _dataframe
-if modules_available("xarray"):
-    from ..structures import xarray as _xarray  # noqa: F401
-
-    del _xarray
+register_builtin_serializers()
 
 
 _FILTER_PARAM_PATTERN = re.compile(r"filter___(?P<name>.*)___(?P<field>[^\d\W][\w\d]+)")
 _LOCAL_TZINFO = dateutil.tz.gettz()
 
+# Pragmatic limit on how "wide" a node can be
+# before the server refusing to inline its contents
+INLINED_CONTENTS_LIMIT = 100
+
+# Pragmatic limit on how deep the server will recurse into
+# nodes that request inlined contents
+DEPTH_LIMIT = 20
```

Comment:
This seems very aggressive. Maybe something more like 3-4?

Reply:
This is a hard upper bound meant to protect the server from being crashed by badly designed or buggy Adapters. It is up to Adapters to opt in to this behavior and decide on a reasonable depth.

I have seen small but highly nester NeXus files, deeper than 3–4 layers, that would be reasonable to marshall greedily.

But I think you are right that 20 is too high. I’ll dial it down to around 5, and we can always raise it later.